src/table/virtualized-table/hooks/use-mutable-prop.ts

```diff
@@ -0,0 +1,10 @@
+import { useRef } from 'react';
+
+const useMutableProp = <T>(value: T) => {
+  const mutableRef = useRef<T>(value);
```

Comment:
I don't really understand this. is it that the value is set on every render? That is why you set current to value on the next line?

Reply:
I'm afraid it's a hack.. I've been using this instead of [useEffectEvent](https://beta.reactjs.org/learn/separating-events-from-effects#declaring-an-effect-event), since that is only experimental.